lib/doorkeeper/openid_connect.rb

```diff
@@ -63,5 +63,33 @@ def self.signing_key_normalized
         key.slice(:kty, :kid)
       end
     end
+
+    if defined?(::Doorkeeper::GrantFlow)
+      Doorkeeper::GrantFlow.register(
+        :token,
```

Comment:
Also I need to register it, but it seems to me that when we add `token` to `grant_flows` (here https://github.com/doorkeeper-gem/doorkeeper-openid_connect/blob/master/lib/doorkeeper/openid_connect/response_types_config.rb#L9) we meant `implicit` grant, right? Or correct me please

Reply:
I'm not sure, this was implemented in https://github.com/doorkeeper-gem/doorkeeper-openid_connect/pull/45 and I think the idea was that you might want to enable Implicit Flow for normal OAuth uses, but not for OIDC (or vice versa). Maybe @nashby can shine some light on this? :grinning:

I'm also not sure if `response_type=token` by itself is valid in an OIDC context, the spec has this table at https://openid.net/specs/openid-connect-core-1_0.html#Authentication:

"response_type" value | Flow
-- | --
code | Authorization Code Flow
id_token | Implicit Flow
id_token token | Implicit Flow
code id_token | Hybrid Flow
code token | Hybrid Flow
code id_token token | Hybrid Flow

But maybe we're adding it in https://github.com/doorkeeper-gem/doorkeeper-openid_connect/blob/master/lib/doorkeeper/openid_connect/response_types_config.rb#L6-L18 in case only `implicit_oidc` is enabled but not Doorkeeper's `implicit` flow, to allow the `code token` combination? If we unified `implicit` and `implicit_oidc` this probably wouldn't be necessary.

In any case, it looks like we can remove the definition for `Doorkeeper::Request::Token` here since we're already adding it in https://github.com/doorkeeper-gem/doorkeeper/blob/b71c5afbfb6694cc023e30be84b9018b5280bf0e/lib/doorkeeper/grant_flow.rb#L11-L15, though with the name `:implicit` instead of `:token`?

Reply:
Yep, I just wanted to be sure it's similar grants :)